trust domain alias secure naming e2e test

liwenhao0810 opened the pull request:
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure


Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

liwenhao0810 committed:
trust domain alias secure naming e2e test

liwenhao0810 committed:
trust domain alias secure naming e2e test

liwenhao0810 committed:
add dynamic certs and test options

liwenhao0810 committed:
add dynamic certs and test options

incfly commented on tests/integration/security/trust_domain_alias/trust_domain_alias_client_test.go:
```diff
@@ -0,0 +1,289 @@
+				})
+			}
+
+			verify(t, client, serverNakedBar, scheme.HTTPS, false)
```
Change this line's scheme to `scheme.HTTP` seems would fix the issue on my end.
You can then enable the verify below. And you will consistently get passing for both foo and bar.

liwenhao0810 committed:
move under ca_custom_root test folder

liwenhao0810 committed:
move under ca_custom_root test folder

liwenhao0810 committed:
fix host to address

linsun committed:
Delete reference to old ISTIO_META_PROXY_XDS_VIA_AGENT (#28203)

* update expose istiod

* add https sample

* fix tab

* update host + domain

* fix lint

* fix lint

* tweak host

* fix lint

* use tls port

* name port correctly

* change default to tls

* Update samples/multicluster/expose-istiod.yaml

Co-authored-by: Iris <irisding@apache.org>

* Update samples/multicluster/expose-istiod.yaml

Co-authored-by: Iris <irisding@apache.org>

* Revert "Update samples/multicluster/expose-istiod.yaml"

This reverts commit 7feb46824dfa7a9cbe2e458cafa98679cf36842b.

* Revert "Update samples/multicluster/expose-istiod.yaml"

This reverts commit 98209a03261dfe3af2f88f81dcfbf76f7787c66d.

* use istiod-remote since pilot is still enabled on remote cluster

* loose up on host name

* adding notes

* clean up this in preview profile

Co-authored-by: Iris <irisding@apache.org>

liwenhao0810 committed:
update script

bianpengyuan committed:
Avoid telemetry cluster metadata override (#28171)

* fix cluster metadata override

* test

* fix

* fix

* fix again

* clean

richardwxn committed:
add telemetry test for customize metrics (#27844)

* add test for customize metrics

* address comments

* add remove tag check

* fix test

jwendell committed:
Delete istiod pods on cleanup (#28205)

Otherwise they stay around and can cause other tests to fail.

In a concrete example, deployment "istiod-canary" stays live
and interferes in pilot's TestMultiRevision test, which also
deploys a "istiod-canary", but, since a deployment with that
name already exists, operator doesn't redeploy it, because it's
already there.

howardjohn committed:
Fix HTTPs on HTTP port passthrough (#28166)

* Fix HTTPs on HTTP port passthrough

* Add note

GregHanson committed:
remove 1.7 telemetry filters from charts (#28195)

irisdingbj committed:
use correct env var name (#28217)

Commit:
Align Ingress resource status updates with Ingresses targeted in controller (#28225)

irisdingbj committed:
make istiod-remote depend on base when installation (#28219)

kylebevans committed:
Add remoteIpBlocks functionality to AuthorizationPolicy (#27906)

* create remoteIpBlocks and update ipBlocks for AuthorizationPolicy

 By adding remoteIpBlocks and notRemoteIpBlocks in Source,
 an AuthorizationPolicy can trigger actions based on the original
 client IP address gleaned from the X-Forwarded-For header or the
 proxy protocol. The ipBlocks and notIpBlocks fields have also been
 updated to use direct_remote_ip in Envoy instead of source_ip

* use correct attribute for RemoteIpBlocks

* fix unit tests and add integration tests for remote.ip attribute

* fix notRemoteIp integration test

* initialize headers if it is nil

* Combine remoteIp tests into IngressGateway test and add release note

* add titles to links

* remove unneeded tests

* fix quotes in releasenote, run make gen

* make upgradeNotes a list

howardjohn committed:
Remove deprecated istio-coredns plugin (#28179)

bianpengyuan committed:
make stackdriver test platform agnostic (#28237)

* make stackdriver test platform agnostic

* fix

* clean up

carolynhu committed:
Add Wasm Extension Dashboard (#28209)

* Add WASM Extension Dashboard

* update dashboard

* update dashboard and add cpu/mem

* address review comment

* add excluded

* remove extension dashboard from test allowlist.txt

* update readme

gargnupur committed:
Add test for local rate limiting using Envoy (#27982)

* Add test for local rate limiting using Envoy

Signed-off-by: gargnupur <gargnupur@google.com>

fix

Signed-off-by: gargnupur <gargnupur@google.com>

add yaml

Signed-off-by: gargnupur <gargnupur@google.com>

fix comments

Signed-off-by: gargnupur <gargnupur@google.com>

fix comments

Signed-off-by: gargnupur <gargnupur@google.com>

* fix build

Signed-off-by: gargnupur <gargnupur@google.com>

* fix build

Signed-off-by: gargnupur <gargnupur@google.com>

bianpengyuan committed:
Clean up metadata exchange keys (#28249)

* clean up

* cleanup exchange key

Commit:
Remove unnecessary warning log from ingress status watcher (#28254)

adiprerepa committed:
vm health checking (#28142)

* impl with pilot

* Remove redundant import

* Remove redundant return

* address some concerns

* address more concerns

* Add tests

* fix ci?

* fix ci?

liwenhao0810 committed:
refactor based on comments

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#27786)

yskopets committed:
pilot: GlobalUnicastIP of a model.Proxy should be set to the 1st applicable IP address in the list (#28260)

* pilot: GlobalUnicastIP of a model.Proxy should be set to the 1st applicable IP address in the list

Signed-off-by: Yaroslav Skopets <yaroslav@tetrate.io>

* docs: add release notes

Signed-off-by: Yaroslav Skopets <yaroslav@tetrate.io>

liwenhao0810 committed:
updated comments

carolynhu committed:
Adjust Wasm VMs charts order and Add release note (#28251)

* Adjust Wasm VMs charts order

* add release note

* replace wasm extension dashboard with real ID

shankgan committed:
Issue #27606: Minor Bug fixes, mostly renaming (#28156)

howardjohn committed:
Cleanup ADS tests (#28275)

* Cleanup ADS tests

* fix lint

* fix lint

liwenhao0810 committed:
add build constraints

liwenhao0810 committed:
lint fix

howardjohn committed:
Temporarily skip ratelimit tests (#28286)

To help master/1.8 get to a merge-able state

liwenhao0810 committed:
fixes based on comments
